### DISCUSSION TITLE
Automatically create a release on GitHub?

### DISCUSSION THREAD

**Discussion by gridsystem** [absolute-version/commit-and-tag-version#167] [Q&A] [answered]
Does commit-and-tag-version help with creating a release from a tag it's made, that would include the relevant items from the changelog?

If not is there a suggested way of doing that?

**Comment by TimothyJones** [comment, marked as answer]
Not at this time, no. 

If you're using github, then [release-please](https://github.com/googleapis/release-please) may work for you - although I personally [ran into](https://github.com/googleapis/release-please/issues/1946) some [bugs](https://github.com/googleapis/release-please/issues/2109) with it. However, for the simple use cases that commit-and-tag-version supports, it seems fine.

Alternatively, you could use a script that combines a `--dry-run` to get the changelog with creating a release on github. I did that for another project a few years ago. Here's the [bash script](https://github.com/pact-foundation/pact-js/blob/master/scripts/ci/release.sh#L18) to extract the changelog and do the tagging, and the corresponding [github workflow](https://github.com/pact-foundation/pact-js/blob/master/.github/workflows/publish.yml#L32) that does the actual creation. This script has some downsides - where it's possible for changes to land while the release is being made, resulting in an inconsistent state of publishes. Yout could rework it to avoid that - by moving the actual artefact publishing to be after the git push.

**Reply by gridsystem** [reply]
That's a fantastic response, thank you!

I haven't gone with the release-please route for the project I'm working on as I have used `standard-version` for quite a few things and I like the workflow. I might have a play with it another time.

For now I will have a look at your scripts 👀 

Thanks again :)
